hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/ShutdownClusterMessageTask.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.management;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.MCShutdownClusterCodec;
+import com.hazelcast.client.impl.protocol.codec.MCShutdownClusterCodec.RequestParameters;
+import com.hazelcast.client.impl.protocol.task.AbstractCallableMessageTask;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.internal.management.ManagementCenterService;
+import com.hazelcast.internal.nio.Connection;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.impl.executionservice.ExecutionService;
+
+import java.security.Permission;
+import java.util.concurrent.Future;
+
+import static com.hazelcast.internal.util.ExceptionUtil.peel;
+import static com.hazelcast.internal.util.ExceptionUtil.withTryCatch;
+
+public class ShutdownClusterMessageTask extends AbstractCallableMessageTask<RequestParameters> {
+
+    public ShutdownClusterMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected Object call() throws Exception {
+        ILogger logger = nodeEngine.getLogger(getClass());
+        ExecutionService executionService = nodeEngine.getExecutionService();
+        Future<Void> future = executionService.submit(
+                ExecutionService.ASYNC_EXECUTOR,
+                () -> {
+                    nodeEngine.getClusterService().shutdown();
+                    return null;
+                });
+
+        executionService.asCompletableFuture(future).whenComplete(
+                withTryCatch(
+                        logger,
+                        (empty, error) -> sendResponse(error != null ? peel(error) : null)));
+
+        return null;
+    }
+
+    @Override
+    protected RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return MCShutdownClusterCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return MCShutdownClusterCodec.encodeResponse();
+    }
+
+    @Override
+    public String getServiceName() {
+        return ManagementCenterService.SERVICE_NAME;
+    }
+
+    @Override
+    public Permission getRequiredPermission() {
+        return null;
```

Comment:
Shouldn't we need a special permission to shutdown cluster from client ?

Reply:
The plan was to add a section to reference manual about how to disable certain MC features using `Security Interceptors` (not to be confused with socket interceptors). But, after your comment, I have discussed the permissions with Sancar and we decided that it would be better to add permissions for MC client operations. As we already have merged quite a few client operations before, I will wait for all MC client operations to be merged before working on their permissions. I've  created the following GH issue track it: https://github.com/hazelcast/management-center/issues/2338